### DISCUSSION TITLE
2.2.0 - Поддержка KeenOS 4+

### DISCUSSION THREAD

**Discussion by znetworkx** [keenetic-dev/bypass_keenetic#31] [Announcements]
- Добавлена поддержка KeenOS 4+
- Добавлена проверка версии KeenOS при установке и обновлении
- Обновлены основные скрипты
- Уменьшен размер бота
- Установка, обновление и удаление теперь идет через скрипт
- IP роутера вычисляется автоматически
- При обновлении через скрипт порты берутся из bot_config.py
- Добавлен список VPN в bot_config.py
- Поправлены скрипты для ключей trojan и v2ray
- Исправлена ошибка, если в имени VPN был дефис
- Проверена работа VPN на KeenOS 4+
- Добавлено сообщение перед перезагрузкой мостов
- Добавлено исправление при превышении списка > 4096 символов
- Добавлены информационные смайлы в меню

## What's Changed
* Update changelog for "2.1.9" by @github-actions in https://github.com/ziwork/bypass_keenetic/pull/13
* Update README.md by @ziwork in https://github.com/ziwork/bypass_keenetic/pull/15
* Update README.md by @ziwork in https://github.com/ziwork/bypass_keenetic/pull/16
* Update README.md by @ziwork in https://github.com/ziwork/bypass_keenetic/pull/17

**Full Changelog**: https://github.com/ziwork/bypass_keenetic/compare/2.1.9...2.2.0

<hr /><em>This discussion was created from the release <a href='https://github.com/ziwork/bypass_keenetic/releases/tag/2.2.0'>2.2.0 - Поддержка KeenOS 4+</a>.</em>

**Comment by saenpass** [comment]
Приветствуем Вас уважаемые авторы супер бота для Keenetic. У сообщества в телеграмм канале вопросы по работе с ключами shadowsocks trojan и vmess. Никак не работает через эти протоколы. Только через tor. Подскажите пожалуйста, может какой есть ньюанс для настройки? Помогите пожалуйста, может есть какой ресурс для помощи?
